embedded/sql/engine.go

```diff
@@ -1109,16 +1111,19 @@ func DecodeValue(b []byte, colType SQLValueType) (TypedValue, int, error) {
 		}
 	case IntegerType:
 		{
-			if vlen > 8 {
+			if vlen != 8 {
 				return nil, 0, ErrCorruptedData
 			}
 
-			buff := [8]byte{0}
+			buff := [8]byte{}
 			copy(buff[8-vlen:], b[voff:voff+vlen])
 			v := binary.BigEndian.Uint64(buff[:])
 			voff += vlen
 
-			return &Number{val: v}, voff, nil
+			// map to signed integer space
+			v += math.MaxInt64 + 1
+
+			return &Number{val: int64(v)}, voff, nil
```

Comment:
I would just use a simple cast:

```suggestion
			return &Number{val: int64(v)}, voff, nil
```

Reply:
replied above